lib/datastore.js

```diff
@@ -80,10 +80,37 @@ Datastore.prototype.loadDatabase = function () {
 
 
 /**
+ * Delete any documents from the array of documents that are expired
+ */
+Datastore.prototype._removeExpiredDocuments = function (docs) {
+  var self = this
+    , expiredDocs = []
+    , validDocs = [];
+
+  docs.forEach(function(doc) {
+    if (model.isExpired(doc) || doc.$$deleted === true) {
```

Comment:
Seems to be an unrealistic requirement as a `$$deleted: true` document would've already been removed from all indexes. Removed the second half of the clause from this `if` statement.